site/docs/hosting/deno-deploy.md

```diff
@@ -0,0 +1,67 @@
+# Hosting: Deno deploy
+
+This guide tells you about the ways you can host your grammY bots on [Deno deploy](https://deno.com/deploy).
+
+Please note that this guide is only for Deno users, and you need to have a [GitHub](https://github.com/github.com) account for creating a [Deno deploy](https://deno.com/deploy) account.
+
+Deno deploy is ideal for most simple bots, and you should note that not all Deno features are available for apps running on Deno deploy. For example, there is no filesys on Deno deploy. It's just like the other many microservices, but dedicated for Deno apps.
+
+## Preparing Your Code
+
+1. Make sure that you have a which exports your `Bot` object, so that you can import it later to run it.
+2. Create a file named `mod.ts` or `mod.js`, or actually any name you like (but you should be remembering and using this as the main file to deploy), with the following content:
+
+```ts
+import { Bot, webhookCallback } from "https://deno.land/x/grammy/mod.ts";
+// You might modify this to the correct way to import your `Bot` object.
+import bot from "./bot.ts";
+
+const handleUpdate = webhookCallback(bot, "serveHttp");
+
+const server = Deno.listen({ port: 3000 });
+
+for await (const conn of server) {
+  serveHttp(conn);
+}
+
+async function serveHttp(conn: Deno.Conn) {
+  const httpConn = Deno.serveHttp(conn);
+
+  for await (const requestEvent of httpConn) {
+    if (requestEvent.request.method == "POST") {
+      try {
+        await handleUpdate(requestEvent);
+      } catch (err) {
+        console.error(err);
+      }
+    } else {
+      await requestEvent.respondWith(new Response(undefined, { status: 200 }));
+    }
+  }
+}
```

Comment:
What do you think about using `std/http` in order to simplify this example?

Reply:
I'm not sure. What do you think?
```suggestion
import { serve } from "https://deno.land/std/http/server.ts";
serve(webhookCallback(bot, "std/http"));
```
Can also remove the definition of `handleUpdate`.

Reply:
I don't know much about it. What if someone sent a `GET` request? And is it concurrent?

Reply:
Try opening https://grammydocsbot.deno.dev/. Confer the defintion at https://github.com/grammyjs/website/blob/7151207fa92ae57c0296aa61f89745018b98fb8c/grammydocsbot/bot.ts#L69.

And I don't assume the `std/http` web server in Deno is somehow not making use of asynchronous programming/concurrency.

Reply:
What I would image is problematic with using `serve` is that you basically cannot make sure requests come from Telegram. The other hosting tutorial introduces secret paths. Perhaps we should directly go for `oak` in order to support this?

We can change it do something more low-level as soon as grammY supports secret paths natively. This will be the case once we do a major release because it will be a breaking change.

Reply:
Aren't you getting any error in the console when you send a `GET` request to your app?

Reply:
I'd expect that to happen, but truth is, I don't.

Reply:
It's unfair to do that. You would get `SyntaxError: Unexpected end of JSON input`.

Reply:
The logs don't show any errors, though. I don't know why not.

Either way, I see that `serve` is not optimal (despite me using it in production lol). How would you suggest to proceed?

Reply:
This might be a better use of the `std/http` way:

```suggestion
import { serve } from "https://deno.land/std/http/server.ts"; 

const handleUpdate = webhookCallback(bot, "std/http");

serve(async (req) => {
  if (req.method == "POST") {
    try {
      return await handleUpdate(req);
     } catch (err) {
       console.error(err);
       return new Response();
     }
   }
   
   return new Response();
});
```

Reply:
Good idea, let's go with this